fix(app-builder-lib): macOS packager uses static icon name

sr258 opened the pull request:
The filename of the app icon in macOS is now always 'icon.icns' instead of a derivate of the product name. The reason for this change is that macOS doesn't display icons with non-ASCII characters in their names, which is quite possible in languages other than English.

This fixes one of the issues described here https://github.com/electron-userland/electron-builder/issues/6282

mmaietta commented:
Can you try rebasing this off latest master? Not sure why one of the tests are failing

sr258 commented:
@mmaietta I rebased this and fixed the test snapshots as far as possible. Unfortunately, I can't really run the tests on my dev machine, as I use an M1 processor, which causes lots of tests to fail even though they shouldn't (see #6310). I hope the CI pipeline runs through this time.
